grails-app/domain/org/pih/warehouse/receiving/Receipt.groovy

```diff
@@ -76,5 +76,16 @@ class Receipt implements Serializable, Comparable<Receipt> {
                         id <=> otherReceipt?.id
     }
 
+    List<ReceiptItem> sortReceiptItemsBySortOrder() {
```

Comment:
I think this is fine, but I think a getSortedReceiptItems() - especially if there's only one way to sort - would be best.

Therefore you can just say 
```
shipments*.receipts*.sortedReceiptItems
```
We should also maybe think about pushing the comparator down to the child's compareTo() method and then it becomes
```
shipments*.receipts*.receiptItems.flatten().sort()
```
or in the case of multiple comparators
```
def receiptItems = Collections.sort(shipments*.receipts*.receiptItems.flatten(), ReceiptItem.customComparator1())
```
The one thing that bothers me is the fact that we're going through multiple objects to get these receipt items. Instead, we should have a named query and/or data service method that retrieves and sorts receipt items. But again, that refactoring can wait until later.